snapcraft/internal/repo/apt_sources_manager.py

```diff
@@ -0,0 +1,231 @@
+        2) If path is specified, convert path to a suite entry,
+           ending with "/".
+
+        Relatedly, this assumes all of the error-checking has been
+        done already on the package_repository object in a proper
+        fashion, but do some sanity checks here anyways.
+
+        :returns: True if source configuration was changed.
+        """
+        if (
+            not package_repo.path
+            and not package_repo.components
+            and not package_repo.suites
+        ):
+            suites = ["/"]
+        elif package_repo.path:
+            # Suites denoting exact path must end with '/'.
+            path = package_repo.path
+            if not path.endswith("/"):
+                path += "/"
+            suites = [path]
+        elif package_repo.suites:
+            suites = package_repo.suites
+            if not package_repo.components:
+                raise RuntimeError("no components with suite")
+        else:
+            raise RuntimeError("no suites or path")
+
+        if package_repo.name:
+            name = package_repo.name
+        else:
+            name = re.sub(r"\W+", "_", package_repo.url)
+
+        return self._install_sources(
+            architectures=package_repo.architectures,
+            components=package_repo.components,
+            formats=package_repo.formats,
+            name=name,
+            suites=suites,
+            url=package_repo.url,
+        )
+
+    def _install_sources_ppa(
+        self, *, package_repo: package_repository.PackageRepositoryAptPpa
+    ) -> bool:
+        """Install PPA formatted repository.
+
+        Create a sources list config by:
+        - Looking up the codename of the host OS and using it as the "suites"
+          entry.
+        - Formulate deb URL to point to PPA.
+        - Enable only "deb" formats.
+
+        :returns: True if source configuration was changed.
+        """
+        owner, name = apt_ppa.split_ppa_parts(ppa=package_repo.ppa)
+        codename = os_release.OsRelease().version_codename()
+
+        return self._install_sources(
+            components=["main"],
+            formats=["deb"],
+            name=f"ppa-{owner}_{name}",
+            suites=[codename],
+            url=f"http://ppa.launchpad.net/{owner}/{name}/ubuntu",
+        )
+
+    def install_package_repository_sources(
+        self, *, package_repo: package_repository.PackageRepository,
+    ) -> bool:
+        """Install configured package repositories.
+
+        :returns: True if source configuration was changed.
+        """
+        logger.debug(f"Processing repo: {package_repo!r}")
+        if isinstance(package_repo, package_repository.PackageRepositoryAptPpa):
+            return self._install_sources_ppa(package_repo=package_repo)
+        elif isinstance(package_repo, package_repository.PackageRepositoryApt):
+            return self._install_sources_apt(package_repo=package_repo)
+        else:
```

Comment:
Nitpick: this if-elif-else cascade shouldn't be necessary since we're returning at each satisfied condition, a plain if-if could result in more readable code.

Reply:
Agreed, fixed